server/src/main/java/io/unitycatalog/server/service/TemporaryTableCredentialsService.java

```diff
@@ -3,41 +3,29 @@
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.server.annotation.ExceptionHandler;
 import com.linecorp.armeria.server.annotation.Post;
-import io.unitycatalog.server.exception.BaseException;
-import io.unitycatalog.server.exception.ErrorCode;
 import io.unitycatalog.server.exception.GlobalExceptionHandler;
-import io.unitycatalog.server.model.*;
+import io.unitycatalog.server.model.GenerateTemporaryTableCredential;
+import io.unitycatalog.server.model.TableInfo;
 import io.unitycatalog.server.persist.TableRepository;
-import io.unitycatalog.server.utils.TemporaryCredentialUtils;
+import io.unitycatalog.server.service.credential.CredentialOperations;
 
 @ExceptionHandler(GlobalExceptionHandler.class)
 public class TemporaryTableCredentialsService {
 
   private static final TableRepository TABLE_REPOSITORY = TableRepository.getInstance();
 
+  private final CredentialOperations credentialOps;
+
+  public TemporaryTableCredentialsService(CredentialOperations credentialOps) {
+    this.credentialOps = credentialOps;
+  }
+
   @Post("")
   public HttpResponse generateTemporaryTableCredential(
       GenerateTemporaryTableCredential generateTemporaryTableCredential) {
     String tableId = generateTemporaryTableCredential.getTableId();
-
-    // Check if table exists
-    String tableStorageLocation = "";
     TableInfo tableInfo = TABLE_REPOSITORY.getTableById(tableId);
```

Comment:
I take it, this is gonna throw if the table doesn't exist? or do you need to check that `tableInfo` is not null before proceeding?

Reply:
yeah the repository is currently taking care of throwing if it doesn't exist...